core/templates/dev/head/services/CsrfTokenService.ts

```diff
@@ -15,48 +15,38 @@
 /**
  * @fileoverview Service for managing CSRF tokens.
  */
+
 var oppia = require('AppInit.ts').module;
 
 oppia.factory('CsrfTokenService', [function() {
-  var token = null;
```

Comment:
So if I understand this correctly the token value is now saved in the `tokenPromise` itself, because when the data is returned the promise will immediately resolve.

Reply:
Yup, that's right. The promise will hold onto the value forever, so there's no need to save it separately since we always return a promise anyway.